frontend/src/metabase/search/util/filter-types.ts

```diff
@@ -0,0 +1,23 @@
+import { FC } from "react";
+import { SearchModelType } from "metabase-types/api";
+
+export const SearchFilterKeys = {
+  Type: "type",
+} as const;
+
+export type TypeFilterProps = SearchModelType[];
+
+type SearchFilterPropTypes = {
+  [SearchFilterKeys.Type]: TypeFilterProps;
+};
+
+export type FilterTypeKeys = keyof SearchFilterPropTypes;
+export type SearchFilters = Partial<SearchFilterPropTypes>;
+
+export type SearchFilterComponent<T extends FilterTypeKeys = any> = FC<
```

Comment:
In the code I'll be using for the `SearchFilterModal`, we'll have a list of components that we'll be able to dynamically enable/disable based on whatever booleans we need to use (i.e. verified/unverified). For the type-check to pass, each component in this list will need to follow the `SearchFilterComponent` spec, which only accepts the keys from `SearchFilterKeys`. So, in effect, to declare a `SearchFilterComponent`, you'll need to declare the key and props, which I think will help with consistency and prevent annoying bugs

Reply:
Here's an example of the type filter component declared with `SearchFilterComponent`:
https://github.com/metabase/metabase/blob/6e9c9f7767ef2c220cba5ddb603f6a736c6dae84/frontend/src/metabase/search/components/SearchFilterModal/filters/TypeFilter.tsx#L14-L18

and where the filter components will be declared:

https://github.com/metabase/metabase/blob/6e9c9f7767ef2c220cba5ddb603f6a736c6dae84/frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal.tsx#L16-L18

and then we can use a list of keys here to selectively display filters in a forEach loop:

https://github.com/metabase/metabase/blob/6e9c9f7767ef2c220cba5ddb603f6a736c6dae84/frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal.tsx#L53-L79